BUILD.gn

```diff
@@ -0,0 +1,204 @@
+    "src/core/mac/mac.cpp",
+    "src/core/mac/mac_filter.cpp",
+    "src/core/mac/mac_frame.cpp",
+    "src/core/mac/sub_mac.cpp",
+    "src/core/mac/sub_mac_callbacks.cpp",
+    "src/core/meshcop/announce_begin_client.cpp",
+    "src/core/meshcop/border_agent.cpp",
+    "src/core/meshcop/commissioner.cpp",
+    "src/core/meshcop/dataset.cpp",
+    "src/core/meshcop/dataset_local.cpp",
+    "src/core/meshcop/dataset_manager.cpp",
+    "src/core/meshcop/dataset_manager_ftd.cpp",
+    "src/core/meshcop/dtls.cpp",
+    "src/core/meshcop/energy_scan_client.cpp",
+    "src/core/meshcop/joiner.cpp",
+    "src/core/meshcop/joiner_router.cpp",
+    "src/core/meshcop/leader.cpp",
+    "src/core/meshcop/meshcop.cpp",
+    "src/core/meshcop/meshcop_tlvs.cpp",
+    "src/core/meshcop/panid_query_client.cpp",
+    "src/core/meshcop/timestamp.cpp",
+    "src/core/net/dhcp6_client.cpp",
+    "src/core/net/dhcp6_server.cpp",
+    "src/core/net/dns_client.cpp",
+    "src/core/net/icmp6.cpp",
+    "src/core/net/ip6.cpp",
+    "src/core/net/ip6_address.cpp",
+    "src/core/net/ip6_filter.cpp",
+    "src/core/net/ip6_headers.cpp",
+    "src/core/net/ip6_mpl.cpp",
+    "src/core/net/netif.cpp",
+    "src/core/net/udp6.cpp",
+    "src/core/radio/radio_callbacks.cpp",
+    "src/core/radio/radio_platform.cpp",
+    "src/core/thread/address_resolver.cpp",
+    "src/core/thread/announce_begin_server.cpp",
+    "src/core/thread/announce_sender.cpp",
+    "src/core/thread/child_table.cpp",
+    "src/core/thread/device_mode.cpp",
+    "src/core/thread/energy_scan_server.cpp",
+    "src/core/thread/indirect_sender.cpp",
+    "src/core/thread/key_manager.cpp",
+    "src/core/thread/link_quality.cpp",
+    "src/core/thread/lowpan.cpp",
+    "src/core/thread/mesh_forwarder.cpp",
+    "src/core/thread/mesh_forwarder_ftd.cpp",
+    "src/core/thread/mesh_forwarder_mtd.cpp",
+    "src/core/thread/mle.cpp",
+    "src/core/thread/mle_router.cpp",
+    "src/core/thread/network_data.cpp",
+    "src/core/thread/network_data_leader.cpp",
+    "src/core/thread/network_data_leader_ftd.cpp",
+    "src/core/thread/network_data_local.cpp",
+    "src/core/thread/network_diagnostic.cpp",
+    "src/core/thread/panid_query_server.cpp",
+    "src/core/thread/router_table.cpp",
+    "src/core/thread/src_match_controller.cpp",
+    "src/core/thread/thread_netif.cpp",
+    "src/core/thread/topology.cpp",
+    "src/core/utils/channel_manager.cpp",
+    "src/core/utils/channel_monitor.cpp",
+    "src/core/utils/child_supervision.cpp",
+    "src/core/utils/heap.cpp",
+    "src/core/utils/jam_detector.cpp",
+    "src/core/utils/missing_strlcpy.c",
+    "src/core/utils/missing_strlcat.c",
+    "src/core/utils/missing_strnlen.c",
+    "src/core/utils/parse_cmdline.cpp",
+    "src/core/utils/slaac_address.cpp",
+    "src/ncp/hdlc.cpp",
+    "src/ncp/spinel.c",
+    "src/ncp/spinel_decoder.cpp",
+    "src/ncp/spinel_encoder.cpp",
+    "third_party/mbedtls/repo/library/md.c",
```

Comment:
Shall we move the mbedtls sources to a separate BUILD.gn in third_party/mbedtls/?

Reply:
It should be possible to do so. What is the motivation for that ? -   Is it for faster incremental build or something else ?

From faster speed perspective, we're already pretty fast given that it's a small project with limited number of files to build. I tried with a single and split BUILD.gn file that in both cases the runtime is about 5ms for gn step (which is what is speeded up because of parallel processing of BUILD.gn files) .  

I have also checked with tools team for Fuchsia project which uses GN about using a single vs hierarchical structure of BUILD.gn files : "In terms of performance, having separate BUILD.gn files for each directory is preferred because GN processes BULD.gn files in parallel, so by using a single file, you're reducing parallelism. On the other hand, if your project isn't too large and the current performance is fine for you, using a single BUILD.gn at this stage should be fine."

Using multiple files introduce some amount of complexity in BUILD.gn files as we have to properly propagate relevant header/configs. So I would like to avoid that if speed is only motivation for splitting.